src/infrastructure/congestion.rs

```diff
@@ -0,0 +1,36 @@
+use crate:: {
+    Config,
+    net::{NetworkQuality, RttMeasurer},
+    sequence_buffer::{CongestionData, SequenceBuffer},
+};
+
+use std::time::Instant;
+
+/// Type responsible for keeping track of congestion information.
+pub struct CongestionHandler {
+    rtt_measurer: RttMeasurer,
+    rtt: f32,
+    congestion_data: SequenceBuffer<CongestionData>,
+    _quality: NetworkQuality,
+}
+
+impl CongestionHandler {
+    pub fn new(config: &Config) -> CongestionHandler {
+        CongestionHandler {
+            rtt_measurer: RttMeasurer::new(config),
+            congestion_data: SequenceBuffer::with_capacity(<u16>::max_value() as usize),
+            _quality: NetworkQuality::Good,
+            rtt: 0.0,
+        }
+    }
+
+    pub fn process_incoming(&mut self, incoming_seq: u16) {
```

Comment:
Comment please

Reply:
I see why you want that, I left it away because it is not public for the user.

Reply:
done